src/lib.rs

```diff
@@ -832,6 +832,39 @@ pub struct SwapChainOutput {
     detail: <C as Context>::SwapChainOutputDetail,
 }
 
+/// The result of a `SwapChain::get_next_texture` operation.
+pub enum SwapChainResult {
```

Comment:
We should probably rely on Rust's `Result` a bit more here:
```rust
struct SwapChainFrame {
  output: SwapChainOutput,
  suboptimal: bool,
};
enum SwapChainError {
...
}
...
Result<SwapChainFrame, SwapChainError>
```

Reply:
Sounds good.

Reply:
It might be nice to try and have some consistency across the project about what the output of a swapchain is actually called though. Right now you obtain it by asking for a "texture" but it's a "frame" that contains an "output" (which the examples then store in a variable called `frame`).

Reply:
Totally agree, the swapchain terminology is a bit of a mess

Reply:
What do you think about either of these:
- `get_next_frame` returning `SwapChainFrame` containing `SwapChainTexture` containing `TextureView`
- `get_next_texture` returning `SwapChainOutput` containing `SwapChainTexture` containing `TextureView`

Reply:
get_next_frame sounds better!